Fixed issue causing random elements of a quadratic field to always be integral

Noel-Roemmele opened the pull request:
Fixes #30017. Fixed error causing all random elements of a quadratic field to be integral. Bug fix was done by Dave Morris in an old pull request at [public/30017](https://github.com/sagemath/sagetrac-mirror/tree/public/30017). Extra doctest and doctest changes were provided by Samuel Lelièvre in their comments on the pull request.

### :memo: Checklist

<!-- Put an `x` in all the boxes that apply. -->

- [X] The title is concise and informative.
- [X] The description explains in detail what this PR is about.
- [X] I have linked a relevant issue or discussion.
- [X] I have created tests covering the changes.
- [X] I have updated the documentation and checked the documentation preview.

### :hourglass: Dependencies

Commit:
Fixed error where all random elements of a quadratic field were integral. Bug fix was done by Dave Morris in an old
pull request at public/30017. Extra doctest and doctest changes were provided by Samuel Lelièvre in their comments
on the pull request.

fchapoton commented on src/sage/rings/number_field/number_field_element.pyx:
```diff
@@ -889,6 +889,42 @@ cdef class NumberFieldElement(NumberFieldElement_base):
+
+        Random integral elements can also be picked using the random_element
+        method of the number field's "ring of integers" or "maximal order" ::
+            sage: K = QuadraticField(2)
```
please add the missing empty line after ::

fchapoton committed:
add blank line in number_field_element.pyx

fchapoton committed:
add blank lines in number_field_element_quadratic.pyx

Commit:
Possibly fix error in random seeded test in src/sage/rings/tests.py. Might be another error in the tests the is
specific to certain computers as fixed test does not correspond to the test Dave Morris and I discussed.

Commit:
Merge branch '30017RandomQuadraticElementIntegral' of github.com:Noel-Roemmele/sage into 30017RandomQuadraticElementIntegral

Commit:
Revert "Possibly fix error in random seeded test in src/sage/rings/tests.py. Might be another error in the tests the is"

This reverts commit 65c54ced6720ef81e5f84fe0cbb0c058e3b38283.

Commit:
Fixed errors in tests found in src/sage/rings/tests.py.

Noel-Roemmele committed:
Merge branch 'sagemath:develop' into 30017RandomQuadraticElementIntegral

Commit:
Changed :trac: into :issue:.

Noel-Roemmele committed:
Merge branch 'sagemath:develop' into 30017RandomQuadraticElementIntegral
